File: ruby/lib/indox_client.rb
# frozen_string_literal: true

require "net/http"
require "json"
require "uri"

# Thin Indox client — Bearer API key + health probe helper.
module IndoxClient
  class Error < StandardError; end

  OPEN_TIMEOUT = 5
  READ_TIMEOUT = 60

  class Client
    attr_reader :api_key, :base_url

    def initialize(api_key: ENV.fetch("INDOX_API_KEY", ""), base_url: ENV.fetch("INDOX_BASE_URL", "https://indox.org"))
      raise Error, "api_key required" if api_key.to_s.strip.empty?
      @api_key = api_key.strip
      @base_url = base_url.to_s.sub(%r{/*$}, "")
    end

    def get(path)
      uri = URI.join("#{@base_url}/", path.sub(%r{^/}, ""))
      req = Net::HTTP::Get.new(uri)
      req["Authorization"] = "Bearer #{api_key}"
      req["Accept"] = "application/json"
      Net::HTTP.start(
        uri.host,
        uri.port,
        use_ssl: uri.scheme == "https",
        open_timeout: OPEN_TIMEOUT,
        read_timeout: READ_TIMEOUT,
        write_timeout: READ_TIMEOUT
      ) do |http|
        http.request(req)
      end
    end

    def health
      get("/api/v1/health/")
    end
  end
end

File: languages/java/src/main/java/org/indox/IndoxClient.java
package org.indox;

/**
 * Thin Indox client — resolves Bearer API key and base URL for the generated SDK.
 */
public final class IndoxClient {
  private final String apiKey;
  private final String baseUrl;

  public IndoxClient(String apiKey, String baseUrl) {
    if (apiKey == null || apiKey.isBlank()) {
      throw new IllegalArgumentException("apiKey required (or set INDOX_API_KEY)");
    }
    this.apiKey = apiKey.trim();
    this.baseUrl = (baseUrl == null || baseUrl.isBlank())
        ? "https://indox.org"
        : baseUrl.replaceAll("/+$", "");
  }

  public static IndoxClient fromEnv() {
    String key = System.getenv().getOrDefault("INDOX_API_KEY", "");
    String base = System.getenv().getOrDefault("INDOX_BASE_URL", "https://indox.org");
    return new IndoxClient(key, base);
  }

  public String getApiKey() {
    return apiKey;
  }

  public String getBaseUrl() {
    return baseUrl;
  }

  public String authorizationHeader() {
    return "Bearer " + apiKey;
  }
}
